frontend_tests/node_tests/message_fetch.js

```diff
@@ -353,7 +353,7 @@ run_test('loading_newer', () => {
         const data = [
             {
                 req: {
-                    anchor: '444',
```

Comment:
This change to the tests causes the test to break (assuming you've addressed the earlier commit, where I had some extensive commentary).

Putting this line back to '444' will get the test passing again.  But now the question is why is the test code still expecting an integer here, instead of the 'first_unread' magic value?  Investigating...

Reply:
I think the test is doing its job here--it's enforcing the fact that `message_fetch.get_frontfill_anchor` is still returning an integer (i.e. `page_params.pointer`):

~~~ js
249 exports.get_frontfill_anchor = function (msg_list) {
250     if (msg_list === home_msg_list) {
251         msg_list = message_list.all;
252     }
253
254     const last_msg = msg_list.last();
255
256     if (last_msg) {
257         return last_msg.id;
258     }
259
260     return page_params.pointer;
261 };
~~~

So I think you back out your change to this line of the test to get this commit passing, but we'll want to talk about when (and if) you intend to change `get_frontfill_anchor`.

Reply:
Hmm.  I think maybe what we actually want to do is make `get_frontfill_anchor` be `"oldest"` and `get_backfill_anchor` be `"newest"` if there are really no messages?  Not sure it really matters, having 0 messages doesn't happen often and it's likely what's most important is just not crashing.

Reply:
It's vice versa, I think, but yup.

Reply:
I think we actually want it the way I described; otherwise you're fetching the empty set of messages.

Actually, thinking about this more, should this case exist at all?  

It seems like we should never be in a situation where our message list is empty and we're running the "we scrolled to the bottom and need to fetch more" code.

Reply:
So instead of returning `page_params.pointer` for the hypothetical-but-probably-impossible case we should instead maybe have a `blueslip.error` call and return `undefined`?  And then let the caller check for `undefined` and abort any fetches?